### ISSUE TITLE
Dutch translations field values do not show in some places

### ISSUE THREAD

**Issue by ocrasborn** [Signbank/NGT-signbank#257]
The Dutch translations of the field values for Compounds/Compound Parts do not show, see screen shot. Didn't detect this problem in other pull-downmenus/

<img width="780" alt="screenshot 2016-12-18 18 19 23" src="https://cloud.githubusercontent.com/assets/10242207/21294798/9c4ace44-c54e-11e6-8bd8-28a9b762ac80.png">

<img width="725" alt="screenshot 2016-12-18 18 21 56" src="https://cloud.githubusercontent.com/assets/10242207/21294803/f1cdf968-c54e-11e6-8261-582ad77983f1.png">

**Comment by ocrasborn** [comment]
Update: I now see that the translations do not appear at all in pull-down menus in the Dutch search form. Did something get broken somewhere?

**Comment by Woseseltops** [comment]
I think this never worked; I only ever added 'translation' to field choice dropdown menus that appear when you click a red editable line in the detail view, and never thought about the rest. I can tell because it's quite a difficult thing to solve within Django (choices themselves are not supposed to be dynamic within Django, let alone translated), and I never had to think about this problem before. It looks like it's going to work, though, I'm about halfway.

For later reference, this is about 4 related situations:
* The morphology role text
* The morphology role editable line
* The dropdown you get when editing a new morphology item
* All 'regular' field choice fields in the search panel (might consist of more subcategories)

**Comment by Woseseltops** [comment]
Done: 
* The morphology role text
* The morphology role editable line
* The dropdown you get when editing a new morphology item

I see a lot of code in that file that translates field choices, though, and for this issue I added even more. I would like to spend some time to make it 1 function with parameters. After that, I go on with translating stuff on the search page.

**Comment by Woseseltops** [comment]
Done:
* Refactored the solution (more elegant, easier to read and maintain)
* I discovered that you also get a non-translated field choice temporarily when you change the morphology role; fixed that as well (easy after the refactorings).

To do:
* I discovered that you can no longer add sequential morphology... needs fixing.
* Translations on the whole search page.

**Comment by Woseseltops** [comment]
Also fixed:

* I discovered that you can no longer add sequential morphology... needs fixing.


As I expected, it was caused by the recent changes in this issue. Pfff... I can imagine it doesn't look like it, but this is incredibly hard and time consuming to get to right. First, I try to find a solution that does what I want, and then I try to come up with an architecture so my solution still more or less makes makes sense to other programmers.

**Comment by Woseseltops** [comment]
Got it working in the search panel as well! I implemented it so that every field defined in the settings gets its choices automatically translated. I used the opportunity to make the drop-down menus prettier (= in the Bootstrap style we use for the rest of SignBank) as well. The morphology search fields are not in the settings, so they still need a manual treatment.

Later addition: turns out this breaks the search functionality again, so I temporarily disabled it again. Note to myself: the problem is that untouched search fields are set as 0 in the GET variables.

**Comment by Woseseltops** [comment]
Fixed!

**Comment by Woseseltops** [comment]
Ooh, almost forgot morphology. I extended the system to the morphology dropdowns, it now works there too. 

However, when testing I noticed also that there are more search fields in the lower panels that do not get translations. I think these would require a very different type of solution (the problems handled in this issue are all about translating field choices) and so a new issue would make things a bit cleaner. @ocrasborn, do we want to create that issue, or do we think these fields are not used enough to justify spending time on trying to get them translated?
